atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/LockWatchValueScopingCacheImpl.java

```diff
@@ -261,6 +260,12 @@ private synchronized void assertNoSnapshotsMissing(Multimap<Sequence, StartTimes
         }
     }
 
+    private Optional<ValueCacheSnapshot> getSnapshotForSequence(Sequence sequence) {
+        synchronized (this) {
+            return snapshotStore.getSnapshotForSequence(sequence);
+        }
+    }
```

Comment:
could be refactored, but as written may make the synchronization more obvious. Either way works for me
```suggestion
    private synchronized Optional<ValueCacheSnapshot> getSnapshotForSequence(Sequence sequence) {
          return snapshotStore.getSnapshotForSequence(sequence);
    }
```